pkg/cluster/info.go

```diff
@@ -174,3 +174,16 @@ func (c *Info) OperatorNamespace() string {
 
 	return constants.OperatorNamespace
 }
+
+func (c *Info) GetBrokers(namespace string) (*v1alpha1.BrokerList, bool, error) {
```

Comment:
There’s no need for a boolean here; if the function can’t return any results, make it return `nil`.

```suggestion
func (c *Info) GetBrokers(namespace string) (*v1alpha1.BrokerList, error) {
```

Reply:
Also for convenience, we can return `[]v1alpha1.Broker` since `BrokerList.Items` is what callers will access anyway (it's the only field in `BrokerList`).

Reply:
> There’s no need for a boolean here; if the function can’t return any results, make it return `nil`.

The bool is returned so that the calling function can use `if found` logic. Ofcourse, we can do `if len(Brokers)` logic but I find the former more readable.

Reply:
You’re returning the same information in two different forms, which is generally speaking not a good idea — if the behaviour changes in future, you can end up with diverging handling in different code paths (depending on whether the code uses `len(Brokers)` or `found`).

Reply:
The caller has to check `len(brokers)` anyway. For the case where there's no error returned, ie `return brokerList, true, nil`,  the `brokerList` will contain an empty `Items` array if no brokers exist so retuning a `found` bool doesn't really add any value.